### DISCUSSION TITLE
how to get the number of elements in Channel.fromPath or Channel.fromFilePairs

### DISCUSSION THREAD

**Discussion by ChenDepp** [nextflow-io/nextflow#5102] [Q&A]
hello everyone
   how to get the number of elements in Channel.fromPath or Channel.fromFilePairs.
   I want to obtain sample data and make subsequent logical judgments.
waiting for your reply
thanks

**Comment by bentsherman** [comment]
Use the `count` operator:

```groovy
Channel.fromPath( '*.fastq' ).count().view { n -> "number of elements: ${n}" }
```

**Reply by ChenDepp** [reply]
the operator count I know it,but I want assign it to variable. This way I can use if else structure

**Reply by bentsherman** [reply]
Using channels in if-else statements is not supported right now, though we are looking into it. For now you will need to use operators like `map`, `subscribe`, `view`, etc to use the count:
```groovy
Channel.fromPath( '*.fastq' ).count().map { n ->
  if( n > 0 ) {
    // ...
  }
  else {
    // ...
  }
}
```
